app.py

```diff
@@ -225,6 +240,9 @@ def main():
     start = time.time()
     prepare_metadata()
     end = time.time()
+    if error_raised:
+        logging.info("Failed to prepare metadata, stopping web server")
+        raise SystemExit
```

Comment:
you could log the errors here for completeness, but it's already been raised

Reply:
Yeah the error gets logged automatically by the requests functions so it would be redundant here. (the error actually gets thrown and logged for every thread, so it's really obvious)

Reply:
ah okay. cool.